servers/services/src/main/java/org/projectnessie/services/impl/TreeApiImpl.java

```diff
@@ -913,57 +922,38 @@ public <R> R getEntries(
                 entries, super::startAccessCheck, getServerConfig().accessChecksBatchSize()) {
               @Override
               protected Set<Check> checksForEntry(KeyEntry entry) {
+                if (!filterPredicate.test(entry)) {
```

Comment:
WDYT about making `AuthzPaginationIterator` extend `FilteringPaginationIterator`? Then, the filter can only be applied once (now twice - on lines 925 and 936).